src/NHSD.BuyingCatalogue.Solutions.Persistence/Repositories/SolutionRepository.cs

```diff
@@ -92,18 +89,8 @@ FROM dbo.Solution
         /// <param name="id">The ID of a <see cref="ISolutionResult"/>.</param>
         /// <param name="cancellationToken">A token to notify if the task operation should be cancelled.</param>
         /// <returns>A task representing an operation to retrieve a <see cref="ISolutionResult"/> matching the specified ID.</returns>
-        public async Task<ISolutionResult> ByIdAsync(string id, CancellationToken cancellationToken)
-        {
-            var solutions = (await dbConnector.QueryAsync<SolutionResult>(ByIdSql, cancellationToken, new { id })).ToList();
-            ISolutionResult solution = solutions.FirstOrDefault();
-            if (solutions.Count > 0)
-            {
-                solution.FrameworkShortNames = new List<string>();
-                solution.FrameworkShortNames = solutions.Select(x => x.FrameworkShortName).ToList();
-            }
-
-            return solution;
-        }
+        public async Task<ISolutionResult> ByIdAsync(string id, CancellationToken cancellationToken) =>
+            (await dbConnector.QueryAsync<SolutionResult>(ByIdSql, cancellationToken, new { id })).FirstOrDefault();
```

Comment:
We should use `SingleOrDefault()` here as a solution ID should be unique.

Reply:
I can apply `SingleOrDefault()` with `DISTINCT` in query, as there can be one solution in multiple `Framework`. 
But in that case  `IsFoundation` column should be same having same data even if it is related to multiple `Framework`

Reply:
Updated SQL query with DISTINCT, assuming `IsFoundation` will be same for solution related to multiple Frameworks(i.e. One solution has multiple entry in `dbo.FrameworkSolution` table). 
Updated method to use `SingleOrDefault()`.

Reply:
I understand why you changed this to `FirstOrDefault` now... Given that, I actually think your original version was better –without `DISTINCT` in the query and with `FirstOrDefault` instead of `SingleOrDefault`. Do you mind reverting it to your original version?

Sorry about that, Rahul!

Reply:
Actually, each way has their pros and cons. Let's leave it like this for now.